include/deal.II/numerics/vector_tools_constraints.h

```diff
@@ -286,14 +291,19 @@ namespace VectorTools
 #else
          .ReferenceCell::get_default_linear_mapping<dim, spacedim>()
 #endif
-         ));
+         ),
+    const IndexSet &   refinement_edge_indices = IndexSet(),
```

Comment:
It makes sense to combine the implementation by adding these parameters, but I would prefer to have the user facing interface to be two functions (a new one called *_on_level). They can both call the same, internal function.